framework/src/geomsearch/PenetrationLocator.C

```diff
@@ -127,7 +127,13 @@ PenetrationLocator::detectPenetration()
 void
 PenetrationLocator::reinit()
 {
+  // Delete the PenetrationInfo objects we own before clearing the
+  // map, or we have a memory leak.
+  for (auto & it : _penetration_info)
+    delete it.second;
```

Comment:
Oh, there are *so* many raw pointer -> smart pointer changes I wanted to make while going through this code, but I was afraid that the changes I already made might be enough to break compatibility with some app or another.